tests/dl_normaliser_test.py

```diff
@@ -0,0 +1,129 @@
+    assert normaliser(
+        'https://id.gs1.org/8006/040123451234560102/22/ABCD/10/XYZ') == "https://id.gs1.org/8006/040123451234560102/10/XYZ"
+    assert normaliser(
+        'https://fashion-corp.com/8006/040123451234560102/10/Lot-123/21/ser987') == "https://id.gs1.org/8006/040123451234560102/21/ser987"
+
+
+def test_sub_domains():
+    assert normaliser(
+        'https://gs1.test.example.org/01/9780345418913/10/1223') == "https://id.gs1.org/01/09780345418913/10/1223"
+
+
+def test_qualifier_or_custom_data_attributes_in_query_string():
+    assert normaliser(
+        'https://example.org/01/9780345418913/21/765tz?11=221109') == "https://id.gs1.org/01/09780345418913/21/765tz"
+    assert normaliser(
+        'https://example.org/01/9780345418913/21/765tz?abc=211121') == "https://id.gs1.org/01/09780345418913/21/765tz"
+
+
+def test_other_gs1_keys():
+    assert normaliser('https://id.gs1.org/00/340123453111111115') == "https://id.gs1.org/00/340123453111111115"
+    assert normaliser('https://id.gs1.org/414/4226350800008') == "https://id.gs1.org/414/4226350800008"
+    assert normaliser('https://example.co.uk/party/4226350800008') == "https://id.gs1.org/417/4226350800008"
+    assert normaliser('https://id.gs1.org/414/4280000000002/254/12') == "https://id.gs1.org/414/4280000000002/254/12"
+    assert normaliser('https://id.gs1.org/8003/03870585000552987') == "https://id.gs1.org/8003/03870585000552987"
+    assert normaliser('https://id.gs1.org/8004/0180451111ABC987') == "https://id.gs1.org/8004/0180451111ABC987"
+    assert normaliser('https://id.gs1.org/8018/385888700111111111') == "https://id.gs1.org/8018/385888700111111111"
+    assert normaliser('https://id.gs1.org/8017/440018922222222226') == "https://id.gs1.org/8017/440018922222222226"
+    assert normaliser('https://id.gs1.org/253/4602443000331XYZ') == "https://id.gs1.org/253/4602443000331XYZ"
+    assert normaliser('https://id.gs1.org/8010/0628165987/8011/9876') == "https://id.gs1.org/8010/0628165987/8011/9876"
+    assert normaliser('https://id.gs1.org/255/0811625999996554433') == "https://id.gs1.org/255/0811625999996554433"
+    assert normaliser(
+        'http://fashion.com/itip/040123451234560102/ser/ABC145') == "https://id.gs1.org/8006/040123451234560102/21/ABC145"
+
+
+def test_non_existent_key_ai():
+    assert normaliser('http://example.org/123/4012345ABC987/456/4711') is None
+    assert normaliser('http://id.gs1.org/99/9780345418913') is None
+    assert normaliser('http://id.gs1.org/01/9780345418913/99/utfgf') is None
+
+
+def test_ordinary_web_uri():
+    assert normaliser('http://example.de/abc/9780345418913') is None
+
+
+def test_incorrect_syntax():
+    assert normaliser('http://example.de/01/97803') is None
+    assert normaliser('https://id.gs1.org/8006/04012345123456/22/ABCD/10/XYZ') is None
+    # the following must not throw
+    assert normaliser('42') is None
+    assert normaliser(42) is None
+    assert normaliser('Hello World!') is None
```

Comment:
@RalphTro here

Reply:
Hi @Echsecutor , apart from the slight adjustment (see below: upui EPC URI), we can merge this from my POV!